pcdsdevices/interface.py

```diff
@@ -1175,12 +1177,36 @@ def movement(scale, direction):
 
 class AbsProgressBar(ProgressBar):
     """Progress bar that displays the absolute position as well."""
+    def __init__(self, status_objs, **kwargs):
+        self._last_position = None
+        self._name = None
+        self._no_more = False
+        super().__init__(status_objs, **kwargs)
+
+        # Extra call when status is complete
+        for i, obj in enumerate(status_objs):
+            if not obj.done:
+                obj.add_callback(functools.partial(self._status_cb, i))
```

Comment:
What's the significance of passing in the index? Is this just what `super()`/`ProgressBar` is expecting?

Reply:
Yes, that's all it is.

Reply:
I think this part should be written slightly differently, I realized that there is a potential race condition here, and that the indices are not guaranteed to be in this order for multiples statuses... But the index does need to be passed in.

Reply:
OK, I see. 👍